Add setup Wizard ready step

yscik opened the pull request:
Fixes #3045 

### Changes proposed in this Pull Request

* Add UI for Ready step.
  * Added an extra *"Join Sensei LMS's Mailing List!"* heading for the sign-up section that seemed to need it.
  * The sign-up still feels a bit buried, I tried out an alternative design where it has its separate card, see screenshot.
* Add an endpoint providing Mailchimp configuration and administrator e-mail.

### Testing instructions

* Everything should behave as described in the #3045 

<!--
Helpful tips for screenshots:
https://en.support.wordpress.com/make-a-screenshot/
-->
### Screenshot / Video
<img width="678" alt="image" src="https://user-images.githubusercontent.com/176949/81968493-be0d1300-961c-11ea-978f-d96bf8129e2a.png">

With import actions:

<img width="678" alt="Screenshot 2020-05-14 at 19 48 06" src="https://user-images.githubusercontent.com/176949/81968302-80a88580-961c-11ea-8d6d-98129db57c76.png">

#### With separate card for signup:
<img width="678" alt="Screenshot 2020-05-14 at 19 49 23" src="https://user-images.githubusercontent.com/176949/81968304-81411c00-961c-11ea-8d24-214457ab4a39.png">

donnapep commented:
Looks good! A couple small things:

- Change _Join Sensei LMS's Mailing List!_ to _Join our mailing list_
- Clicking _Create a Course_ should go to `/post-new.php?post_type=course`. Let's drop them right into the _Add New Course_ page.

yscik commented on assets/onboarding/ready/index.jsx:
```diff
@@ -0,0 +1,78 @@
+/**
+ * Ready step for Setup Wizard.
+ */
+export const Ready = () => {
```
Tab saved! :) 
5feebf3

yscik commented on assets/onboarding/ready/index.jsx:
```diff
@@ -0,0 +1,78 @@
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { MailingListSignupForm } from './mailinglist-signup-form';
+import { formatString } from '../../tests-helper/format-string';
```
Moved to a `helpers` folder in e108edb

yscik commented:
Updated to latest master and integrated the new store. 
Added some tests for the Ready component.
Also added a new `/ready` endpoint to mark the end of the setup wizard (that removes the notice). It's called when the features step is finished.

donnapep commented on assets/onboarding/ready/index.jsx:
```diff
@@ -0,0 +1,73 @@
+				<H>{ __( `Join our mailing list`, 'sensei-lms' ) }</H>
+				<p>
+					{ __(
+						`We're here for you — Get tips, product updates, and inspiration straight to your mailbox.`,
```
Should be "get" vs. "Get" as this is the continuation of the same sentence.

yscik replied:
Updated in 562d2bdc.

renatho commented on assets/onboarding/features/index.jsx:
```diff
@@ -113,6 +114,7 @@ const Features = () => {
 	};
 
 	const goToNextStep = () => {
+		completeSetupWizard();
```
I'm wondering if it would make more sense doing it while loading the ready step, using the same `submitStep` from the `useSetupWizardStep`. WDYT?

We could also make a check if the ready step is already completed to don't submit if the user navigate back in the steps and go to the ready again. This check, maybe we could do while working in this issue: #3135

donnapep replied:
> I'm wondering if it would make more sense doing it while loading the ready step,

This makes sense to me, as it would more easily account for the possibility of injecting a new step between Features and Ready.

yscik replied:
I'll remove this from here and add the call in #3135, as it can be better handled with some state management around steps. Running it every time the Ready component is mounted is a bit messy. - 9bab849b

renatho replied:
Should we remove also the related actions?

yscik replied:
The functionality and the REST API endpoint is still part of this PR, so I think it's frontend action can stay too.

renatho commented on assets/onboarding/ready/index.test.js:
```diff
@@ -0,0 +1,49 @@
+} ) );
+
+describe( '<Ready />', () => {
+	afterEach( () => {} );
```
I think we could remove this `afterEach`, right?

yscik replied:
Yup, 2434b0c3

jom commented:
Sorry for the post-merge comment, but is there any plan to add a way to get out of the wizard on this step without clicking create course? I wanted to get out of the wizard and that seemed to be the only path.

In previous versions of Woo's wizard, there would be a link below/outside of the white box to let me leave. Now, Woo takes me back into normal WP Admin and displays a modal over WooCommerce Dashboard page, which is also fine because I can quickly choose where I navigate next.

yscik commented:
We could turn off the fullscreen mode here, so WP-Admin is fully accessible again.

donnapep commented:
> We could turn off the fullscreen mode here, so WP-Admin is fully accessible again.

Oohhh interesting. I'd love to see how that would feel.
